app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -1342,8 +1342,7 @@ private void saveDataToDisk(boolean exit, boolean complete, String updatedSaveNa
         // save current answer; if headless, don't evaluate the constraints
         // before doing so.
         boolean wasScreenSaved =
-                saveAnswersForCurrentScreen(headless ? DO_NOT_EVALUATE_CONSTRAINTS : EVALUATE_CONSTRAINTS,
-                        complete, headless);
+                saveAnswersForCurrentScreen(DO_NOT_EVALUATE_CONSTRAINTS, complete, headless);
```

Comment:
This is the main save code-path, so it is sort of scary to be making a change like this late in the QA cycle.

Also seems counter-intuitive to make this change, but we are saved by additional constraint checked when the form starts saving: https://github.com/dimagi/commcare-odk/blob/master/app/src/org/commcare/tasks/SaveToDiskTask.java#L94